cmd/sriov/main.go

```diff
@@ -219,6 +219,11 @@ func cmdDel(args *skel.CmdArgs) error {
 		return nil
 	}
 
+	// Verify VF ID existence.
+	if _, err := utils.GetVfid(netConf.DeviceID, netConf.Master); err != nil {
+		return fmt.Errorf("cmdDel() error obtaining VF ID: %q", err)
```

Comment:
if VF disappeared for some reason, do we  need to fail ? or should we return nil here?

container runtime/ multus will retry delete command, do we expect a different outcome the next time around ?

Reply:
We should return an error to let user know that the VF is gone. And no, we don't expect a different outcome next time.

The defer of the Del command cleans up the cached netconf when the err var is nil. This new section follows the same logic as the one for ResetVFConfig and ReleaseVF (inside the if for dpdkMode) which consists of overwriting the err variable. By doing this, the returned error is logged (crio/kubelet logs), the defer cleans up the netconf cache and the next retry of the Del command returns nil (first if clause) because the cache is not longer there. So, we managed to communicate the actual error but the Del command does not stay in a loop.

Reply:
ah, i missed the `:=` here which redefines the `err` in this inner scope.

a horrible obscurity IMO which we need to get rid of. hopefully once we add support for logging in cni.